LanguageFeatures/Augmentation-libraries/augmented_expression_A01_t28.dart

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// - Augmenting getters: Within an augmenting getter `augmented` invokes the
+///   augmented getter and evaluates to its return value. If augmenting a
+///   variable with a getter, this will invoke the implicitly induced getter
+///   from the augmented variable declaration.
+///
+/// @description Checks that it is a compile-time error if an augmenting getter
+/// has a metadata named `augmented`.
```

Comment:
I think this is a little bit to strong, it should only be an error for the metadata `@augmented()` to occur _inside_ a declaration which is augmenting (that is: which has the modifier `augment`). It is not an error (see line 27) for an augmenting declaration to have such metadata, as long as it's not inside a construct that has the modifier `augment`.

The reason why it's fine to have "the same thing in a different place" is that the problem is the confusion which can be caused by the _word_ `augmented` occurring in a scope where `augmented`/`augmented()` can be an invocation of an augmented method, getter, setter, or initializing expression. There is nothing wrong with the _meaning_ of `augmented` in these cases, they would have been perfectly OK if their name had been different.

Reply:
Works - thank you!